src/L1/rollup/ScrollChain.sol

```diff
@@ -816,39 +852,35 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
             if (onlyOne) break;
         }
 
+        if (parentBatchHash != lastBatchHash) {
```

Comment:
Would be great to add a comment here why and how this could happen:
- faulty batch producer
- unexpected `parentBatch` in case commit transactions get reordered 
- anything else?